python/beeai_framework/backend/types.py

```diff
@@ -75,13 +82,26 @@ def from_chunks(cls, chunks: list[Self]) -> Self:
     def merge(self, other: Self) -> None:
         self.messages.extend(other.messages)
         self.finish_reason = other.finish_reason
+
+        if self.cost is not None and other.cost is not None:
```

Comment:
You don't handle cases where `self.cost` is `None` and `other.cost` is not `None`